doc/subsystems/test/sanitycheck.rst

```diff
@@ -385,6 +385,63 @@ extra_sections: <list of extra binary sections>
     extra, unexpected sections in the Zephyr binary unless they are named
     here. They will not be included in the size calculation.
 
+harness: <list of harnesses>
+    A list of harnesses needed to run the tests sucessfully. This can be
+    as simple as a loopback wiring or a complete hardware test setup for
+    sensor and IO testing. Usually pertains to external dependency domains
+    but can be anything such as console, sensor, net, keyboard, or Bluetooth.
+
+harness_config: <harness configuration options>
+    Extra harness configuration options to be used to select a board and/or
+    for handling generic Console with regex matching. Config can announce
+    what features it supports. This option will enable the test to run on
+    only those platforms that fulfill this external dependency.
+
+    The following options are currently supported:
+
+    type: <one_line|multi_line> (required)
+        Depends on the regex string to be matched
+
+    regex: <expression> (required)
+        Any string that the particular test case prints to confirm test
+        runs as expected.
+
+    ordered: <True|False> (default False)
+        Check the regular expression strings in orderly or randomly fashion
+
+    repeat: <integer>
+        Number of times to validate the repeated regex expression
+
+    fixture: <expression>
+        Specify a test case dependency on an external device(e.g., sensor),
+        and identify boards that fulfills this dependency. It depends on
```

Comment:
and identify setups that fulfil this dependency...

Reply:
Modified